Makefile

```diff
@@ -50,18 +50,17 @@ IMAGE:=$(REGISTRY)/node-problem-detector:$(TAG)
 ENABLE_JOURNALD?=1
 
 # TODO(random-liu): Support different architectures.
-BASEIMAGE:=alpine:3.4
+# The debian-base:0.3 image built from kubernetes repository is based on Debian Stretch.
+# It includes systemd 232 with support for both +XZ and +LZ4 compression.
+# +LZ4 is needed on some os distros such as COS.
+BASEIMAGE:=gcr.io/google-containers/debian-base-amd64:0.3
 
 # Disable cgo by default to make the binary statically linked.
 CGO_ENABLED:=0
 
-# NOTE that enable journald will increase the image size.
 ifeq ($(ENABLE_JOURNALD), 1)
 	# Enable journald build tag.
 	BUILD_TAGS:=-tags journald
-	# Use fedora because it has newer systemd version (229) and support +LZ4. +LZ4 is needed
-	# on some os distros such as GCI.
-	BASEIMAGE:=fedora
```

Comment:
> I feel like we should use smaller image if we don't need journald support.

A few points I have against that:

1. Every relevant Linux distribution uses systemd nowadays, so the point of supporting no journal is kind of moot to me...

2. We already use debian-base-amd:0.3 as base layer for other images (such as fluentd-gcp, etc.) so reusing it might actually reduce overall disk usage since they're all sharing from the same base.

3. Supporting two builds means this needs more testing, it's likely we might end up breaking one but not the other, which might go unnoticed as one of them is seldom or never used... Let's try to reduce overall complexity.

Having said that, if you really insist, I think it's OK to keep using Alpine for no journald... Let me know if you really want to keep Alpine for the non-journal case.

Reply:
I agree with all @filbranden 's points.

Reply:
> Every relevant Linux distribution uses systemd nowadays, so the point of supporting no journal is kind of moot to me...

We have no journal support because we were still using containerVM at that time.

I still think we should support no journal, because journal support:
* need `lib-systemd-dev` to build;
* will make the binary non-static.

I still prefer making it possible to have a simple statically linked binary for users who don't use journal, e.g. people may use this in `LinuxKit`.

As for the base image, as long as `BASEIMAGE` is over-writable, I'm fine with always using `gcr.io/google-containers/debian-base-amd64:0.3` as the default base image to reduce the maintenance overhead.

Reply:
I think this PR still does what you suggested... It still keeps the ENABLE_JOURNALD variable and sets build tags appropriately in those cases. So, good to merge?

Reply:
Yeah :)